pkg/docker/config/config.go

```diff
@@ -85,27 +90,36 @@ func GetAuthentication(sys *types.SystemContext, registry string) (string, strin
 	}
 
 	dockerLegacyPath := filepath.Join(homedir.Get(), dockerLegacyHomePath)
-	var paths []string
-	pathToAuth, err := getPathToAuth(sys)
+	paths := []authPath{}
+	pathToAuth, lf, err := getPathToAuth(sys)
 	if err == nil {
-		paths = append(paths, pathToAuth)
+		paths = append(paths, authPath{path: pathToAuth, legacyFormat: lf})
 	} else {
 		// Error means that the path set for XDG_RUNTIME_DIR does not exist
 		// but we don't want to completely fail in the case that the user is pulling a public image
 		// Logging the error as a warning instead and moving on to pulling the image
 		logrus.Warnf("%v: Trying to pull image in the event that it is a public image.", err)
 	}
-	paths = append(paths, filepath.Join(homedir.Get(), dockerHomePath), dockerLegacyPath)
+	homeDirDockerCFG := authPath{
+		path:         dockerLegacyPath,
+		legacyFormat: true,
+	}
+	homeDirConfigJSON := authPath{
+		path:         filepath.Join(homedir.Get(), dockerHomePath),
+		legacyFormat: false,
+	}
+	paths = append(paths, homeDirConfigJSON, homeDirDockerCFG)
```

Comment:
Please use the same order for the `paths = ` build and for declaring the two variables above — or, maybe even better, inline it without adding the extra variables:
```go
paths = append(paths,
    authPath{path:…, legacyFormat: false},
    authPath{path:…. legacyFormat: true}
)
```

Reply:
part of next push